ui/app/models/ssh/ca-config.js

```diff
@@ -5,16 +5,49 @@
 
 import Model, { attr } from '@ember-data/model';
 import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import { withModelValidations } from 'vault/decorators/model-validations';
 
+const validations = {
+  publicKey: [
+    {
+      validator(model) {
+        const { publicKey, privateKey, generateSigningKey } = model;
+        // if generateSigningKey is false, use the second validation to check if both public and private keys are set
+        return (publicKey && privateKey) || (!publicKey && !privateKey) || !generateSigningKey ? true : false;
+      },
+      message: 'Public Key and Private Key are both required if one of them is set.',
+    },
+  ],
+  generateSigningKey: [
+    {
+      validator(model) {
+        const { publicKey, privateKey, generateSigningKey } = model;
+        if (!generateSigningKey && (!publicKey || !privateKey)) {
+          return false;
+        }
+        return true;
+      },
+      message: 'Public Key and Private Key are both required if Generate Signing Key is false.',
```

Comment:
```suggestion
      message: 'Error: Provide a public and private key or set "Generate Signing Key" to true.',
```
Suggest writing in active voice and being more direct about how to solve the error